src/org/mirah/jvm/compiler/constructor_cleanup.mirah

```diff
@@ -55,11 +55,19 @@ class ConstructorCleanup < SimpleNodeVisitor
   def visitNodeList(node, arg)
     node.size.times do |i|
       child = node.get(i)
-      unless child.kind_of?(ClassDefinition)
-        return child.accept(self, arg)
-      end
+      res = child.accept(self, arg)
+      return res if res
     end
   end
+  
+  def visitClassDefinition(node, arg)
```

Comment:
Could you add a comment to this and the closure definition one noting that class definitions inside constructors shouldn't be cleaned up?